### PR TITLE
ViaVersion Integration

### PR DESCRIPTION
**Pull request by Ph4i1ur3**
Added ViaVersion.jar in included lib directory, updated pom.xml, created bridge and new property for PlayerTags: <PlayerTag.viaversion>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -349,6 +349,13 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>us.myles</groupId>
+            <artifactId>ViaVersion</artifactId>
+            <version>2.1.3</version>
+            <scope>system</scope>
+            <systemPath>${basedir}/lib/ViaVersion.jar</systemPath>
+        </dependency>
```
Comment (mcmonkey4eva) [inline pom.xml]: Dependency list is alphabetical order.

File: src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    PlayerTag player;
+
+    @Override
+    public String getAttribute(Attribute attribute) {
+        if (attribute == null) {
+            return null;
+        }
+
+        //Getting the ViaVersion API object
+        ViaAPI api = Via.getAPI();
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java]: This should be INSIDE the attribute `if` block, not outside.

File: src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    PlayerTag player;
+
+    @Override
+    public String getAttribute(Attribute attribute) {
+        if (attribute == null) {
+            return null;
+        }
+
+        //Getting the ViaVersion API object
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java]: Unnecessary/redundant comment

File: src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        ViaAPI api = Via.getAPI();
+
+
+        // <--[tag]
+        // @attribute <PlayerTag.viaversion>
+        // @returns ElementTag(Number)
+        // @description
+        // Returns the protocol version number of the player's client.
+        // See https://wiki.vg/Protocol_version_numbers as a reference list.
+        // @Plugin Depenizen, ViaVersion.
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java]: Remove the period from this line

File: src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        //Getting the ViaVersion API object
+        ViaAPI api = Via.getAPI();
+
+
+        // <--[tag]
+        // @attribute <PlayerTag.viaversion>
+        // @returns ElementTag(Number)
+        // @description
+        // Returns the protocol version number of the player's client.
+        // See https://wiki.vg/Protocol_version_numbers as a reference list.
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java]: Use `<@link url ...>`

**Comment by Ph4i1ur3** [comment]
Issues have been corrected.

**Comment by Ph4i1ur3** [comment]
Erm.. 1 more change needed... I suck at the alphabet...

**Comment by Ph4i1ur3** [comment]
Hah... Yep. 1 more..

File: src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    PlayerTag player;
+
+    @Override
+    public String getAttribute(Attribute attribute) {
+        if (attribute == null) {
+            return null;
+        }
+
+
+
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java]: Too many blank lines (remove all but one)

File: src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        // <--[tag]
+        // @attribute <PlayerTag.viaversion>
+        // @returns ElementTag(Number)
+        // @description
+        // Returns the protocol version number of the player's client.
+        // See <@link url https://wiki.vg/Protocol_version_numbers as a reference list.>
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/properties/viaversion/ViaVersionPlayerProperties.java]: the `>` goes at the end of the link not the end of the line

**Comment by mcmonkey4eva** [comment]
Looks good now.

Thanks for contributing ❤️

**Pull request merged**